major_event_log/feeds.py

```diff
@@ -1,32 +1,156 @@
+
+            if self.feed['author_link'] is not None:
+                handler.addQuickElement('uri', self.feed['author_link'])
+
+            handler.endElement('author')
+
+        if self.feed['subtitle'] is not None:
+            handler.addQuickElement('subtitle', self.feed['subtitle'])
+
+        for cat in self.feed['categories']:
+            handler.addQuickElement('category', '', {'term': cat})
+
+        if self.feed['feed_copyright'] is not None:
+            handler.addQuickElement('rights', self.feed['feed_copyright'])
+
+
+class MajorEventLogFeed(PaginatedAtom1FeedMixin, Atom1Feed):
     """Custom Atom Feed for Event objects.
 
-    The only change between this custom feed and the base Atom Feed
-    is that this feed specifies its mime_type as 'application/xml'.
+    The only changes between this custom feed and the base Atom Feed
+    are that this feed specifies its mime_type as 'application/xml',
+    pagination is added, and the 'alternate' link is removed.
     """
+
     mime_type = 'application/xml'
 
 
-class LatestEventsFeed(Feed):
-    # Custom Atom feed.
+class PaginatedFeedMixin(object):
+    """Feed mixin to enable pagination."""
+
+    paginator = None
+    page = 1
+    items_per_page = 10
+    page_field = 'page'
+
+    def setup_paginator(self, request, items):
+        """Instantiates the paginator object and attaches
+        it to self.
+
+        This needs to be called from `get_object` in order to receive
+        the request object.
+        """
+        self.page = request.GET.get(self.page_field, 1)
+        self.paginator = Paginator(items, self.items_per_page)
+
+    def get_current_page(self):
+        """Get the current page of the Paginator."""
+        return self.paginator.page(self.page)
+
+    def get_page_kwargs(self):
+        """Returns the keyword arguments for the page links.
+
+        This is intended to be called from `feed_extra_kwargs`.
+        """
+        page = self.get_current_page()
+        kwargs = {}
+
+        kwargs.setdefault('page_field', self.page_field)
+        if page.has_next():
+            kwargs.setdefault('next_page', page.next_page_number())
+
+        if page.has_previous():
+            kwargs.setdefault('prev_page', page.previous_page_number())
+
+        kwargs.setdefault('last_page', str(self.paginator.num_pages))
+        return kwargs
+
+
+class LatestEventsFeed(PaginatedFeedMixin, Feed):
     feed_type = MajorEventLogFeed
-    # Required tags by the Atom feed.
-    title = "PREMIS Major Event Log"
-    link = "/major-event-log/"
-    subtitle = "10 most recent major PREMIS events."
+    title = 'PREMIS Major Event Log'
+    link = reverse_lazy('major-event-log:feed')
+    subtitle = '10 most recent major PREMIS events.'
+    author_name = 'Major Event Log'
+    author_link = 'http://digital2.library.unt.edu/name/nm0005293/'
+    page_field = 'p'
+
+    def get_object(self, request):
+        events = Event.objects.order_by('-date')
```

Comment:
Should we order by 'date' of the event as it is or  'entry_modified' or 'entry_created'? You could create an event with a date in long in the past, and it will get dropped near the end of the feed.